packages/twenty-front/src/modules/object-record/record-board/hooks/internal/useSetRecordBoardColumns.ts

```diff
@@ -0,0 +1,48 @@
+import { useRecoilCallback } from 'recoil';
+
+import { useRecordBoardStates } from '@/object-record/record-board/hooks/internal/useRecordBoardStates';
+import { RecordBoardColumnDefinition } from '@/object-record/record-board/types/RecordBoardColumnDefinition';
+import { isDeeplyEqual } from '~/utils/isDeeplyEqual';
+
+export const useSetRecordBoardColumns = (recordBoardId?: string) => {
+  const { scopeId, getColumnIdsState, columnsFamilySelector } =
+    useRecordBoardStates(recordBoardId);
+
+  const setRecordBoardColumns = useRecoilCallback(
+    ({ set, snapshot }) =>
+      (columns: RecordBoardColumnDefinition[]) => {
+        const currentColumns = snapshot
+          .getLoadable(getColumnIdsState())
+          .getValue();
+
+        const columnIds = columns.map(({ id }) => id);
```

Comment:
I prefer this pattern but if it's not widely used you can ignore!
```suggestion
        const columnIds = columns.map(column => column.id)
```